engine/api/database/common.go

```diff
@@ -0,0 +1,44 @@
+package database
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-gorp/gorp"
+	"github.com/lib/pq"
+
+	"github.com/ovh/cds/sdk"
+)
+
+// IDsToQueryString returns a comma separated list of given ids.
+func IDsToQueryString(ids []int64) string {
+	res := make([]string, len(ids))
+	for i := 0; i < len(ids); i++ {
+		res[i] = fmt.Sprintf("%d", ids[i])
+	}
+	return strings.Join(res, ",")
+}
+
+// Insert value in given db.
+func Insert(db gorp.SqlExecutor, i interface{}) error {
+	err := db.Insert(i)
+	if e, ok := err.(*pq.Error); ok && e.Code == ViolateUniqueKeyPGCode {
+		err = sdk.NewError(sdk.ErrConflict, e)
+	}
+	return sdk.WithStack(err)
+}
+
+// Update value in given db.
+func Update(db gorp.SqlExecutor, i interface{}) error {
+	_, err := db.Update(i)
+	return sdk.WithStack(err)
+}
+
+// Delete value in given db.
+func Delete(db gorp.SqlExecutor, i interface{}) error {
+	_, err := db.Delete(i)
+	return sdk.WithStack(err)
+}
+
+// And returns givens query strings joined with AND.
+func And(qs ...string) string { return fmt.Sprintf("(%s)", strings.Join(qs, " AND ")) }
```

Comment:
I'm trying something in this PR, I want to write only two dao funcs to query a resource then give the criteria. I want to prevent the creation of multiple func like GetAllByID or GetAllByIDAndSlug...
That's why I need to create WHERE closes dynamically with funcs like this one. This is something we should discuss to find the better way of doing.

Reply:
removed